params.py

```diff
@@ -0,0 +1,372 @@
+                return ret_val
+        else:
+            raise ValueError("Unsupported hyperparameter distribution type")
+
+    def sample(self) -> Any:
+        """Randomly sample a value from the distribution of this HyperParameter."""
+        return self.ppf(random.uniform(0.0, 1.0))
+
+    def _to_config(self) -> Tuple[str, Dict]:
+        config = dict(value=self.value)
+        return self.name, config
+
+
+class HyperParameterSet(list):
+    def __init__(self, items: List[HyperParameter]):
+        """A set of HyperParameters.
+
+        >>> hp1 = HyperParameter('a', {'values': [1, 2, 3]})
+        >>> hp2 = HyperParameter('b', {'distribution': 'normal'})
+        >>> HyperParameterSet([hp1, hp2])
+
+        Args:
+            items: A list of HyperParameters to construct the set from.
+        """
+
+        for item in items:
+            if not isinstance(item, HyperParameter):
+                raise TypeError(
+                    f"each item used to initialize HyperParameterSet must be a HyperParameter, got {item}"
+                )
+
+        super().__init__(items)
+        self.searchable_params = [
+            param for param in self if param.type != HyperParameter.CONSTANT
+        ]
+
+        self.param_names_to_index = {}
+        self.param_names_to_param = {}
+
+        for ii, param in enumerate(self.searchable_params):
+            self.param_names_to_index[param.name] = ii
+            self.param_names_to_param[param.name] = param
+
+    @classmethod
+    def from_config(cls, config: Dict):
+        """Instantiate a HyperParameterSet based the parameters section of a SweepConfig.
+
+        >>> sweep_config = {'method': 'grid', 'parameters': {'a': {'values': [1, 2, 3]}}}
+        >>> hps = HyperParameterSet.from_config(sweep_config['parameters'])
+
+        Args:
+            config: The parameters section of a SweepConfig.
+        """
+        hpd = cls(
+            [
+                HyperParameter(param_name, param_config)
+                for param_name, param_config in sorted(config.items())
+            ]
+        )
+        return hpd
+
+    def to_config(self) -> Dict:
+        """Convert a HyperParameterSet to a SweepRun config."""
+        return dict([param._to_config() for param in self])
+
+    def denormalize_vector(self, X: npt.ArrayLike) -> List[List[Any]]:
+        """Converts a list of vectors [0,1] to values in the original space.
+
+        Args:
+            X: 2-d array-like. Array of normalized parameter values. Rows are instances,
+            columns are parameters.
+        Returns:
+            List of denormalized instances (lists) of parameter vectors.
+        """
+        columns = []
+        for ii, param in enumerate(self.searchable_params):
+            vec = param.ppf(X[:, ii])
+            try:
+                vec = vec.tolist()
+            except AttributeError:
+                pass
+            columns.append(vec)
+        return [list(row) for row in zip(*columns)]
```

Comment:
denormalize_vector should be covered.

Reply:
or removed/commented out if not active anymore

Reply:
this is no longer used